File: DSA/Anisha_Parveen/Day_13_Ques_2.cpp
#include <bits/stdc++.h>
using namespace std;

bool isPalindrome(string &numberString, int start, int end)
{
    if (start >= end)
        return true;
    if (numberString[start] != numberString[end])
        return false;
    return isPalindrome(numberString, start + 1, end - 1);
}

bool palindrome(int number)
{
    string numberString = to_string(number);
    return isPalindrome(numberString, 0, numberString.size() - 1);
}

int main()
{

    int number;
    cout << "Enter a number: ";
    cin >> number;

    bool ans = palindrome(number);
    if (ans)
        cout << "YES";
    else
        cout << "NO";
}

File: DSA/Asmi_Garg/day13_ques2.java
public class Palindrome {

    public static void main(String[] args) {
        int number = 121;
        if (isPalindrome(number)) {
            System.out.println("YES");
        } else {
            System.out.println("NO");
        }
    }

    public static boolean isPalindrome(int number) {
        String str = Integer.toString(number);
        return isPalindromeHelper(str, 0, str.length() - 1);
    }

    private static boolean isPalindromeHelper(String str, int left, int right) {
        if (left >= right) {
            return true;
        }
        if (str.charAt(left) != str.charAt(right)) {
            return false;
        }
        return isPalindromeHelper(str, left + 1, right - 1);
    }
}
